Bio/PDB/MMCIFParser.py

```diff
@@ -67,6 +67,49 @@ def get_structure(self, structure_id, filename):
 
     # Private methods
 
+    def _mmcif_get(self, key, dict, deflt):
+        if key in dict:
+            rslt = dict[key][0]
+            if "?" != rslt:
+                return rslt
+        return deflt
+
+    def _get_mmcif_header_dict(self, md):
+        dict = {
+            "name": "",
+            "head": "",
+            "idcode": "",
+            "deposition_date": "",
+            "structure_method": "",
+            "resolution": 0.0,
+        }
+
+        dict["idcode"] = self._mmcif_get("_entry.id", md, None)
+        if not dict["idcode"]:
+            dict["idcode"] = self._mmcif_get("_exptl.entry_id", md, None)
+        if not dict["idcode"]:
+            dict["idcode"] = self._mmcif_get("_struct.entry_id", md, None)
+
+        dict["name"] = self._mmcif_get("_struct.title", md, None)
+
+        dict["head"] = self._mmcif_get("_struct_keywords.pdbx_keywords", md, None)
+        if not dict["head"]:
+            dict["head"] = self._mmcif_get("_struct_keywords.text", md, None)
+
+        dict["deposition_date"] = self._mmcif_get(
+            "_pdbx_database_status.recvd_initial_deposition_date", md, None
+        )
+
+        dict["structure_method"] = self._mmcif_get("_exptl.method", md, None)
+
+        dict["resolution"] = float(self._mmcif_get("_refine.ls_d_res_high", md, 0.0))
+        if 0.0 == dict["resolution"]:
```

Comment:
I'd rather swap the order in if-statements to be more readable: `if x == 0.0` instead of `if 0.0 == x`.

Reply:
removed with other changes.

Reply:
Thanks!